### PR TITLE
[FIX] load_data: update to version 17.0

### PR DESCRIPTION
**Pull request by royle-viindoo**
changed on 17.0
(https://github.com/odoo/odoo/commit/5bf1207c8cb7575fba4501aff4932c2cc952bf9f)

### CODE REVIEW

**Comment by legalsylvain** [comment]
I dont get it.

**Comment by pedrobaeza** [comment]
On v16-, you call the method `load_data(cr, ...)`. On v17+, you call it `load_data(env, ...)`.

**Comment by legalsylvain** [comment]
OK ! 

You propose to replace 
```
def load_data(cr, module_name, filename, idref=None, mode="init"):
```

by 
```
def load_data(env_or_cr, module_name, filename, idref=None, mode="init"):
```
 
Adding a comment to say that an cr is required until 16 and an env is required since 17, right ?


Personnaly, I prefer to keep the same signature ``def load_data(cr, module_name, filename, idref=None, mode="init"):``

Then, add : 
```
env_or_cr = api.Environment(cr, SUPERUSER_ID, {}) if version_info[0] >= 17 else cr
```
then simply replace ```tools.convert_xml_import(cr, module_name, fp, idref, mode=mode)```
by ```tools.convert_xml_import(env_or_cr, module_name, fp, idref, mode=mode)```

But both works, so it's maybe a matter of taste. I think that it can maybe generate more errors to change the type of the argument, depending of the version of the module that call the function.

**Comment by royle-viindoo** [comment]
> OK !
> 
> You propose to replace
> 
> ```
> def load_data(cr, module_name, filename, idref=None, mode="init"):
> ```
> 
> by
> 
> ```
> def load_data(env_or_cr, module_name, filename, idref=None, mode="init"):
> ```
> 
> Adding a comment to say that an cr is required until 16 and an env is required since 17, right ?
> 
> Personnaly, I prefer to keep the same signature `def load_data(cr, module_name, filename, idref=None, mode="init"):`
> 
> Then, add :
> 
> ```
> env_or_cr = api.Environment(cr, SUPERUSER_ID, {}) if version_info[0] >= 17 else cr
> ```
> 
> then simply replace `tools.convert_xml_import(cr, module_name, fp, idref, mode=mode)` by `tools.convert_xml_import(env_or_cr, module_name, fp, idref, mode=mode)`
> 
> But both works, so it's maybe a matter of taste. I think that it can maybe generate more errors to change the type of the argument, depending of the version of the module that call the function.



> OK !
> 
> You propose to replace
> 
> ```
> def load_data(cr, module_name, filename, idref=None, mode="init"):
> ```
> 
> by
> 
> ```
> def load_data(env_or_cr, module_name, filename, idref=None, mode="init"):
> ```
> 
> Adding a comment to say that an cr is required until 16 and an env is required since 17, right ?
> 
> Personnaly, I prefer to keep the same signature `def load_data(cr, module_name, filename, idref=None, mode="init"):`
> 
> Then, add :
> 
> ```
> env_or_cr = api.Environment(cr, SUPERUSER_ID, {}) if version_info[0] >= 17 else cr
> ```
> 
> then simply replace `tools.convert_xml_import(cr, module_name, fp, idref, mode=mode)` by `tools.convert_xml_import(env_or_cr, module_name, fp, idref, mode=mode)`
> 
> But both works, so it's maybe a matter of taste. I think that it can maybe generate more errors to change the type of the argument, depending of the version of the module that call the function.

@legalsylvain Thank you for your solution. I find it very interesting and low risk. I have fixed it according to your solution.

**Comment by pedrobaeza** [comment]
Why I was talking to pass `env` is for the same reason as stated in the Odoo commit: to re-use the environment and avoid to regenerate it each time, and to share the context. Don't you think it's worth?

**Comment by royle-viindoo** [comment]
> Why I was talking to pass `env` is for the same reason as stated in the Odoo commit: to re-use the environment and avoid to regenerate it each time, and to share the context. Don't you think it's worth?

@pedrobaeza I have updated it according to your suggestion, please review it

File: openupgradelib/openupgrade.py [hunk 1]
```diff
@@ -329,21 +337,21 @@ def load_data(cr, module_name, filename, idref=None, mode="init"):
         if ext == ".csv":
             noupdate = True
             tools.convert_csv_import(
-                cr, module_name, pathname, fp.read(), idref, mode, noupdate
+                env_or_cr, module_name, pathname, fp.read(), idref, mode, noupdate
             )
         elif ext == ".yml":
             yaml_import(cr, module_name, fp, None, idref=idref, mode=mode)
```
Comment (pedrobaeza) [inline openupgradelib/openupgrade.py:343]: This one hasn't been changed, isn't it?

Comment (royle-viindoo) [reply openupgradelib/openupgrade.py:345]: Has been changed at [5bf1207c8cb7575fba4501aff4932c2cc952bf9f](https://github.com/odoo/odoo/commit/5bf1207c8cb7575fba4501aff4932c2cc952bf9f)

Comment (StefanRijnhart) [reply openupgradelib/openupgrade.py:345]: `yaml_import` was removed from Odoo after 11 anyway.

**Review by StefanRijnhart — APPROVED** [review]

File: openupgradelib/openupgrade.py (hunk 1 above)
Comment (pedrobaeza) [reply openupgradelib/openupgrade.py:345]: I know, but right now, the code will fail in 11-, as `cr` has no value for that case. Assign previously `cr = env_or_cr`, or directly put here `env_or_cr`.

Comment (royle-viindoo) [reply openupgradelib/openupgrade.py:346]: you are right, I added more at line 310

**Review by pedrobaeza — APPROVED** [review]
Thanks for the patience attending our comments.

**Pull request merged**